docs/diagnostics/PX1037.md

```diff
@@ -0,0 +1,46 @@
+# PX1037
+This document describes the PX1037 diagnostic.
+
+## Summary
+
+| Code   | Short Description                                 | Type  | Code Fix    | 
+| ------ | ------------------------------------------------- | ----- | ----------- | 
+| PX1037 | Unbound DAC field cannot be used in a key declaration  | Warning (ISV Level 3: Informational) | Available | 
+
+## Diagnostic Description
+The diagnostic checks the fields which are used in a primary, foreign, or unique key declaration. The key declaration cannot contain unbound fields.
+The warning is shown only for DACs that have the `PXCacheName` or `PXPrimaryGraph` attribute.
```

Comment:
The behavior of DAC keys analysis was changed. Now we divide checks in two groups - the important basic checks which apply to a wider scope of DACs and more specific code style related checks which apply to a lesser scope of correctly declared not fully unbound DACs with `PXCacheName` or `PXPrimaryGraph` attribute.

The check for unbound DAC keys in DAC applies to all non abstract DACs